### ISSUE TITLE
--mapdir flag

### ISSUE THREAD

**Issue by willemneal** [wasmerio/wasmer#458]
Great stuff!  I've been implementing my own Wasi library in [Assemblyscript](https://github.com/WebAssemblyOS/wasmos) and have had some initial successes using wasmer!

### Motivation

It would be nice for end users if their initial directory path was root, "/" instead of either a fullpath or ".".

Wasmtime has this option so that you can map an external pathname to an internal one.

e.g. `--mapdir=/tmp:/var/tmp`

### Proposed solution

Add command line option and open directories on the host with the path left of the ":" and name it with the left half in the runtime.

[Currently this is done here.](https://github.com/wasmerio/wasmer/blob/88124d926eaa45e2dfd6424f2b86cd3e728fa6ff/lib/wasi/src/state.rs#L193)

Wasmtime does it [here](https://github.com/CraneStation/wasmtime/blob/3ae7c60b1395fe25971f683828241e7fac8cb40b/src/wasmtime.rs#L135)

### Alternatives

One alternative is to just interpret each path as a mapped directory if there is a ":" in the name.  This seems like the faster way, but it would be better if the API reflected the difference.

**Comment by MarkMcCaskey** [comment]
Thanks -- we're glad to hear that!  This is a great idea and we agree it's super nice to have!

I've started work on it in #451 actually, but am blocked by myself (well, due to mysterious CI issues) setting up tests (#457).  I wanted to have more confidence that I wasn't breaking core features before shipping/continuing development on it.

**Comment by willemneal** [comment]
Awesome! Look forward to it!
